packages/sitecore-jss-react/src/components/VisitorIdentification.test.tsx

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable no-unused-expressions */
+import React from 'react';
+import { expect } from 'chai';
+import { mount } from 'enzyme';
+
+import { VisitorIdentification, resetEmittedVI } from './VisitorIdentification';
+import { SitecoreContextReactContext } from './SitecoreContext';
+
+describe('<VisitorIdentification />', () => {
+  const defaultContext = { visitorIdentificationTimestamp: 123 };
+
+  const mockContext = (context) => {
+    return {
+      context: context,
+    };
+  };
+
+  afterEach(() => {
+    const script = document.head.getElementsByTagName('script')[0];
+    const meta = document.head.getElementsByTagName('meta')[0];
+    if (script) {
+      document.head.removeChild(script);
+    }
+    if (meta) {
+      document.head.removeChild(meta);
+    }
+    resetEmittedVI();
+  });
+
+  it('should render in <head>', () => {
+    const rendered = mount(
+      <SitecoreContextReactContext.Provider value={mockContext(defaultContext)}>
+        <VisitorIdentification />
+      </SitecoreContextReactContext.Provider>
+    );
+    expect(document.head.childElementCount).to.be.equal(2);
+    const script = document.head.getElementsByTagName('script')[0];
+    const meta = document.head.getElementsByTagName('meta')[0];
+    expect(script).to.not.be.equal(undefined);
+    expect(meta).to.not.be.equal(undefined);
+    expect(script.src).to.equal('/layouts/system/VisitorIdentification.js');
+    expect(rendered.html()).to.be.null;
+  });
+
+  it('should render script with defer when defer is enabled', async () => {
+    const rendered = mount(
+      <SitecoreContextReactContext.Provider value={mockContext(defaultContext)}>
+        <VisitorIdentification defer={true} />
+      </SitecoreContextReactContext.Provider>
+    );
+    const script = document.getElementsByTagName('script')[0];
+    expect(script).to.not.be.equal(undefined);
+    expect(script.defer).to.equal(true);
```

Comment:
We should check the `defer` default somewhere. Probably fine to just add an expectation to the `should render in <head>` test?

Reply:
Good idea. I've added the extra check.